### PR TITLE
Add support for non-JSON response types in http_common

### PR DESCRIPTION
**Pull request by gordonwang0**
Adds `mod response`, which provides various response types for the HTTP server. Previously, the server only supported JSON response types, but others (zip, chunked) will be needed as part of the edged refactor.

### CODE REVIEW

File: http-common/src/server.rs [hunk 1]
```diff
@@ -340,31 +326,54 @@ impl Error {
...
+
+    pub fn zip(
+        status_code: hyper::StatusCode,
+        size: usize,
+        body: Vec<u8>,
+    ) -> hyper::Response<hyper::Body> {
+        let res = hyper::Response::builder().status(status_code);
+
+        let res = res
+            .header(hyper::header::CONTENT_ENCODING, "zip")
```
Comment (gordonwang0) [inline http-common/src/server.rs:369]: This header is probably wrong, but it's what the current edged uses. So I left it.

Comment (arsing) [reply http-common/src/server.rs:369]: Yes, it's wrong. It should be `deflate`.

It should be fine to change it - it was already a bad value so no self-respecting client library would be depending on it anyway.

**Review by arsing — CHANGES_REQUESTED** [review]

File: http-common/src/server.rs (hunk 1 above)
Comment (gordonwang0) [reply http-common/src/server.rs]: Ok, fixed it.

File: http-common/src/server.rs [hunk 2]
```diff
@@ -340,31 +326,54 @@ impl Error {
...
-    body: Option<&impl serde::Serialize>,
-) -> hyper::Response<hyper::Body> {
-    let res = hyper::Response::builder().status(status_code);
-    // `res` is consumed by both branches, so this cannot be replaced with `Option::map_or_else`
-    //
-    // Ref: https://github.com/rust-lang/rust-clippy/issues/5822
-    #[allow(clippy::option_if_let_else)]
-    let res = if let Some(body) = body {
+pub mod response {
+    pub fn empty() -> hyper::Response<hyper::Body> {
```
[the remark this answers is not in the document]
Comment (gordonwang0) [reply http-common/src/server.rs]: Fixed.

**Review by arsing — APPROVED** [review]

**Pull request merged**